### PR TITLE
Replace some getPointerElementType calls

### PR DESCRIPTION
**Pull request by Flakebi**
getPointerElementType will go away once llvm switches to opaque
pointers, so start replacing some of the uses with alternatives.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit f6046787219637f7afd53f079b224190564ffe6e
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 39f8940199e60c44d4211cf8165dfd12876316fa</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 2483d46fa27c30502e497ea169ee53b142e9fa06</li>
<li>SPVGEN: dev fa39b110650bc72fd7fda54af621e69306036888</li>
<li>XGL: dev 36c5637649089bf65d42c36b431f47c170b57276</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev eb9d3792bdfda0b6fb94af6bd9989cad51842403</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/201585)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 38010/67195 (56.6%)</li>
<li>Failed: 0/67195 (0.0%)</li>
<li>Not Supported: 29185/67195 (43.4%)</li><li>Warnings: 0/67195 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 37848/67195 (56.3%)</li>
<li>Failed: 0/67195 (0.0%)</li>
<li>Not Supported: 29347/67195 (43.7%)</li><li>Warnings: 0/67195 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 39069/67195 (58.1%)</li>
<li>Failed: 0/67195 (0.0%)</li>
<li>Not Supported: 28126/67195 (41.9%)</li><li>Warnings: 0/67195 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/SystemValues.cpp [hunk 1]
```diff
@@ -298,22 +298,22 @@ Value *ShaderSystemValues::getGsVsRingBufDesc(unsigned streamId) {
 
 // =====================================================================================================================
 // Get pointers to emit counters (GS)
-ArrayRef<Value *> ShaderSystemValues::getEmitCounterPtr() {
+std::pair<llvm::Type *, ArrayRef<llvm::Value *>> ShaderSystemValues::getEmitCounterPtr() {
```
Comment (amdrexu) [inline lgc/patch/SystemValues.cpp:301]: Remove "llvm::" prefix in .cpp files

Comment (amdrexu) [inline lgc/patch/SystemValues.cpp:301]: Remove "llvm::" prefix in .cpp files

File: lgc/patch/SystemValues.cpp [hunk 2]
```diff
@@ -400,10 +401,11 @@ Value *ShaderSystemValues::getStreamOutBufDesc(unsigned xfbBuffer) {
 
 // =====================================================================================================================
 // Get stream-out buffer table pointer
-Instruction *ShaderSystemValues::getStreamOutTablePtr() {
+std::pair<Type *, Instruction *> ShaderSystemValues::getStreamOutTablePtr() {
   assert(m_shaderStage == ShaderStageVertex || m_shaderStage == ShaderStageTessEval ||
          m_shaderStage == ShaderStageCopyShader);
 
+  auto *ty = ArrayType::get(FixedVectorType::get(Type::getInt32Ty(*m_context), 4), MaxTransformFeedbackBuffers);
```
Comment (amdrexu) [inline lgc/patch/SystemValues.cpp:408]: ty -> streamOutTableTy

Comment (amdrexu) [inline lgc/patch/SystemValues.cpp:408]: ty -> streamOutTableTy

File: lgc/patch/SystemValues.cpp [hunk 3]
```diff
@@ -298,22 +298,22 @@ Value *ShaderSystemValues::getGsVsRingBufDesc(unsigned streamId) {
 
 // =====================================================================================================================
 // Get pointers to emit counters (GS)
-ArrayRef<Value *> ShaderSystemValues::getEmitCounterPtr() {
+std::pair<llvm::Type *, ArrayRef<llvm::Value *>> ShaderSystemValues::getEmitCounterPtr() {
   assert(m_shaderStage == ShaderStageGeometry);
+  auto *i32Ty = Type::getInt32Ty(*m_context);
```
Comment (amdrexu) [inline lgc/patch/SystemValues.cpp:303]: i32Ty -> emitCounterTy

Comment (amdrexu) [inline lgc/patch/SystemValues.cpp:303]: i32Ty -> emitCounterTy

**Review by amdrexu — COMMENTED** [review]
LGTM. Maybe we need an internal change as well since some internal codes need it.

**Review by amdrexu — APPROVED** [review]

**Comment by Flakebi** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 96c5ba6ca7a3b0f7afc7cdda14961a23c5e07d61
<details><summary>CTS tests (Failed: 0/201273)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 38006/67091 (56.6%)</li>
<li>Failed: 0/67091 (0.0%)</li>
<li>Not Supported: 29085/67091 (43.4%)</li><li>Warnings: 0/67091 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 37844/67091 (56.4%)</li>
<li>Failed: 0/67091 (0.0%)</li>
<li>Not Supported: 29247/67091 (43.6%)</li><li>Warnings: 0/67091 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 39065/67091 (58.2%)</li>
<li>Failed: 0/67091 (0.0%)</li>
<li>Not Supported: 28026/67091 (41.8%)</li><li>Warnings: 0/67091 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**